tools/commands-cordova.js

```diff
@@ -638,38 +673,64 @@ var ensureCordovaPlugins = function (projectContext, options) {
     }
   });
 
-  if (shouldReinstallPlugins) {
+  if (! _.isEmpty(pluginsFromLocalPath))
+    verboseLog('Reinstalling cordova plugins added from the local path');
+
+  if (shouldReinstallPlugins || ! _.isEmpty(pluginsFromLocalPath)) {
     // Loop through all of the current plugins and remove them one by one until
-    // we have no plugins. It's necessary to loop because we might have
-    // dependencies between plugins.
-    var uninstallAllPlugins = function () {
+    // we have deleted proper amount of plugins. It's necessary to loop because
```

Comment:
Ok, maybe I have not explained it clearly. Before this PR like you said if there was any change (added/remove plugins) we've been deleting all the plugins because it was "safe". This behavior was not changed. Still if you will add/remove any plugins to the project it will delete all of them and install them again just like it did before. 
Since we need to refresh the local plugins (as we do not track changes in their sources) on every build I wanted to reinstall only the local plugins and that is why I needed to modify this function to support this case. On line 711 you can see that I am either reinstalling all of the plugins or just those added from local path so I am never selectively reinstalling a plugin that is added from url or cordova repo - so the "safe" way of ensuring the good state of cordova build was retained.